### PR TITLE
Feature: Generalization of RandomNumberGenerator::random_vec()

### PR DESCRIPTION
**Pull request by reneme**
This is a spin-off from #3195 to harvest the low-hanging fruit. Namely, allowing to use `RNG::random_vec()` with arbitrary contiguous byte buffers.

Example usage (e.g. as desired in #3150):

```C++
// generate a random TLS Session ID
using Session_ID = Botan::Strong<std::vector<uint8_t>, struct Session_ID_>;
const auto random_id = rng.random_vec<Session_ID>(32);

// filling an arbitrary buffer with random bytes
std::array<uint8_t, 32> buffer;
rng.random_vec(buffer);
```

### CODE REVIEW

File: src/lib/utils/concepts.h [hunk 1]
```diff
@@ -12,6 +12,7 @@
 #include <compare>
 #include <concepts>
 #include <type_traits>
+#include <ostream>
```
Comment (randombit) [inline src/lib/utils/concepts.h:15]: Either this was a include left over from debugging, or there is something I'm missing about how these concepts work.

**Review by randombit — APPROVED** [review]
This part is fine

File: src/lib/utils/concepts.h (hunk 1 above)
Comment (reneme) [reply src/lib/utils/concepts.h:15]: TL;DR: include what you use

The include is needed for the `streamable<>` concept. This was not introduced here, but the compiler needed the header when including `<botan/concepts.h>` in some compilation unit that didn't slurp it in through other ways.

Its likely that this particular PR wouldn't have needed it. Nevertheless I left it in for completeness.

Comment (randombit) [reply src/lib/utils/concepts.h:15]: OK I was only looking at the diff and missed the preexisting concept. Yeah that's fine.

**Pull request merged**